process-migration-service/src/main/java/org/kie/processmigration/service/impl/MigrationServiceImpl.java

```diff
@@ -0,0 +1,278 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(MigrationServiceImpl.class);
+
+    private static final List<Integer> QUERY_PROCESS_INSTANCE_STATUSES = Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
+    public static final Integer QUERY_PAGE_SIZE = 100;
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Inject
+    private PlanService planService;
+
+    @Inject
+    private KieService kieService;
+
+    @Inject
+    private SchedulerService schedulerService;
+
+    @Override
+    public Migration get(Long id) throws MigrationNotFoundException {
+        TypedQuery<Migration> query = em.createNamedQuery("Migration.findById", Migration.class);
+        query.setParameter("id", id);
+        try {
+            return query.getSingleResult();
+        } catch (NoResultException e) {
+            throw new MigrationNotFoundException(id);
+        }
+    }
+
+    @Override
+    public List<MigrationReport> getResults(Long id) throws MigrationNotFoundException {
+        Migration m = get(id);
+        TypedQuery<MigrationReport> query = em.createNamedQuery("MigrationReport.findByMigrationId", MigrationReport.class);
+        query.setParameter("id", m.getId());
+        return query.getResultList();
+    }
+
+    @Override
+    public List<Migration> findAll() {
+        return em.createNamedQuery("Migration.findAll", Migration.class).getResultList();
+    }
+
+    @Override
+    @Transactional
+    public Migration submit(MigrationDefinition definition) throws InvalidMigrationException {
+        validateDefinition(definition);
+        Migration migration = new Migration(definition);
+        em.persist(migration);
+        if (ExecutionType.SYNC.equals(definition.getExecution().getType())) {
+            migrate(migration);
+        } else {
+            schedulerService.scheduleMigration(migration);
+        }
+        return migration;
+    }
+
+    @Override
+    @Transactional
+    public Migration delete(Long id) throws MigrationNotFoundException {
+        Migration migration = get(id);
+        em.remove(migration);
+        return migration;
+    }
+
+    @Override
+    @Transactional
+    public Migration update(Long id, MigrationDefinition definition) throws MigrationNotFoundException, ReScheduleException, InvalidMigrationException {
+        validateDefinition(definition);
+        Migration migration = get(id);
+        if (migration == null) {
+            return null;
+        }
+        if (!ExecutionStatus.SCHEDULED.equals(migration.getStatus())) {
+            throw new ReScheduleException("The migration is not scheduled and cannot be re-scheduled");
+        }
+        if (ExecutionType.SYNC.equals(definition.getExecution().getType())) {
+            throw new ReScheduleException("The migration execution type MUST be ASYNC");
+        }
+        migration.setDefinition(definition);
+        em.persist(migration);
```

Comment:
As we discussed on the chat, this can be removed completely since `migration` is already managed because it was fetched from DB in `get()` method. So call to `setDefinition()` will be propagated to DB when committing.